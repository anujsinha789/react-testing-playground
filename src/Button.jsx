import React, { memo } from "react";
import { isEqual } from "lodash";

function Button({ text, onClick }) {
  console.log(text, " button rerendered");
  return <button onClick={onClick}>{text}</button>;
}

export default memo(Button, isEqual);
