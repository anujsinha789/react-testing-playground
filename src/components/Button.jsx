import React, { memo } from "react";

function Button({ text, onClick }) {
  console.log(text, " button rerendered");
  return (
    <button data-testid="counter_btn" onClick={onClick}>
      {text}
    </button>
  );
}

export default memo(Button);
