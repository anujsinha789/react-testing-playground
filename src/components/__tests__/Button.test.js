import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Button from "../Button";

afterEach(() => {
  cleanup(); // runs after every test case and cleans up the render.
});

test("should render button component", () => {
  render(<Button />);
  const Btn = screen.getByTestId("counter_btn");
  expect(Btn).toBeInTheDocument();
});

test("should render text passed as prop", () => {
  render(<Button text="Click me!!" />);
  const Btn = screen.getByTestId("counter_btn");
  expect(Btn).toHaveTextContent("Click me!!");
  expect(Btn).not.toHaveClass(); // Button component should not have any class.
});

test("should match snapshot", () => {
  const tree = renderer.create(<Button text="Click me!!" onClick={() => {}} />);
  console.log(tree);
  expect(tree).toMatchSnapshot();
});
