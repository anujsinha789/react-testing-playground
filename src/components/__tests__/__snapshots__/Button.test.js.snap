// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  data-testid="counter_btn"
  onClick={[Function]}
>
  Click me!!
</button>
`;
